provider/docker/docker.go

```diff
@@ -726,6 +730,41 @@ func (p *Provider) getBasicAuth(container dockerData) []string {
 	return []string{}
 }
 
+func (p *Provider) hasRequestHeaders(container dockerData) bool {
+	label, err := getLabel(container, types.LabelFrontendRequestHeader)
+	return err == nil && len(label) > 0
+}
+
+func (p *Provider) hasResponseHeaders(container dockerData) bool {
+	label, err := getLabel(container, types.LabelFrontendResponseHeader)
+	return err == nil && len(label) > 0
+}
+
+func (p *Provider) getRequestHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendRequestHeader)
+}
+
+func (p *Provider) getResponseHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendResponseHeader)
+}
+
+func parseCustomHeaders(container dockerData, containerType string) map[string]string {
+	customHeaders := make(map[string]string)
+	if label, err := getLabel(container, containerType); err == nil {
+		for _, headers := range strings.Split(label, ",") {
```

Comment:
`Access-Control-Allow-Methods:POST,GET,OPTIONS` wouldn't this header cause this split to fail? maybe we should use a `;` similar to how the rules are separated?

Reply:
```
time="2017-12-01T00:03:00Z" level=warning msg="Could not load header "traefik.frontend.headers.customrequestheaders": [POST], skipping..."
time="2017-12-01T00:03:00Z" level=warning msg="Could not load header "traefik.frontend.headers.customrequestheaders": [OPTIONS], skipping..."
time="2017-12-01T00:03:00Z" level=warning msg="Could not load header "traefik.frontend.headers.customrequestheaders": [DELETE], skipping..."
time="2017-12-01T00:03:00Z" level=warning msg="Could not load header "traefik.frontend.headers.customrequestheaders": [PUT], skipping..."
```

Reply:
@vito-c please open an issue instead of comment on an already merged PR.